tests/components/automation/test_init.py

```diff
@@ -894,3 +894,57 @@ def test_automation_not_trigger_on_bootstrap(hass):
     hass.bus.async_fire('test_event')
     await hass.async_block_till_done()
     assert 'Service not found' in caplog.text
+
+
+async def test_automation_restore_last_triggered_with_initial_state(hass):
+    """Ensure last_triggered is restored, even when initial state is set."""
+    time = dt_util.utcnow()
+
+    mock_restore_cache(hass, (
+        State('automation.hello', STATE_ON),
+        State('automation.bye', STATE_ON, {'last_triggered': time}),
+        State('automation.solong', STATE_OFF, {'last_triggered': time}),
+    ))
+
+    config = {automation.DOMAIN: [{
+        'alias': 'hello',
+        'initial_state': 'off',
+        'trigger': {
+            'platform': 'event',
+            'event_type': 'test_event',
+        },
+        'action': {'service': 'test.automation'}
+    }, {
+        'alias': 'bye',
+        'initial_state': 'off',
+        'trigger': {
+            'platform': 'event',
+            'event_type': 'test_event',
+        },
+        'action': {'service': 'test.automation'}
+    }, {
+        'alias': 'solong',
+        'initial_state': 'on',
+        'trigger': {
+            'platform': 'event',
+            'event_type': 'test_event',
+        },
+        'action': {'service': 'test.automation'}
+    }]}
+
+    await async_setup_component(hass, automation.DOMAIN, config)
+
+    state = hass.states.get('automation.hello')
+    assert state
+    assert state.state == STATE_OFF
+    assert state.attributes.get('last_triggered') is None
```

Comment:
Are you asserting that `last_triggered` is not present in the dictionary or that it's present but with a value of `None`?  This should be made tighter to verify the intended behavior...  i.e. `assert 'last_triggered' not in state.attributes` or `assert state.attributes['last_triggered'] is None`.

Reply:
The `last_triggered` attribute is hardcoded to be returned always in the state attributes of `AutomationEntity`. It will always be there, by default initialized as `None`.

It really won't matter, secondly, this is consistent with all other tests.